generators/app/index.js

```diff
@@ -151,37 +155,49 @@ module.exports = Generator.extend({
           }
         );
       }
+
+      this.composer = options.drupalDistro.loadComposer(this, options);
     },
 
     // While there are write operations in this, other write operations may need
     // to examine the composer.json to determine their own configuration.
     composerJson: function () {
-      // Import any existing composer.json that was saved above.
-      var composer = {};
       var isNewProject = (this.composerOrig == undefined);
-      if (isNewProject) {
-        // Project hasn't been generated yet, so start with composer template.
-        composer = this.fs.readJSON('composer.json');
-      }
-      else {
+      if (!isNewProject) {
         // Use original composer file if project already generated.
-        composer = this.composerOrig;
+        this.composer = this.composerOrig;
       }
 
-      composer.name = options.projectName;
-      composer.description = options.projectDescription;
+      this.composer.name = options.projectName;
+      this.composer.description = options.projectDescription;
       // Allow distros to modify the composer.json.
       if (typeof options.drupalDistro.modifyComposer == 'function') {
         var done = this.async();
-        composer = options.drupalDistro.modifyComposer(this, options, composer, isNewProject, done);
+        var self = this;
```

Comment:
Code I touched where we were passing `this` around or about to I tried to switch to the self convention. TBH I'm not sure if it's needed, but was trying to create some more consistency with what we already had spottily in the codebase. https://alistapart.com/article/getoutbindingsituations